libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1700,21 +1700,33 @@ Type const* TypeChecker::typeCheckTypeConversionAndRetrieveReturnType(
 	vector<ASTPointer<Expression const>> const& arguments = _functionCall.arguments();
 	bool const isPositionalCall = _functionCall.names().empty();
 
-	Type const* resultType = dynamic_cast<TypeType const&>(*expressionType).actualType();
+	Type const* resultType = nullptr;
 	if (arguments.size() != 1)
-		m_errorReporter.typeError(
+		// TODO see if this be made non-fatal
+		m_errorReporter.fatalTypeError(
 			2558_error,
 			_functionCall.location(),
 			"Exactly one argument expected for explicit type conversion."
 		);
 	else if (!isPositionalCall)
-		m_errorReporter.typeError(
-			5153_error,
-			_functionCall.location(),
-			"Type conversion cannot allow named arguments."
-		);
+		m_errorReporter
+			.typeError(5153_error, _functionCall.location(), "Type conversion cannot allow named arguments.");
+	else if (auto userDefinedValueTypeType = dynamic_cast<UserDefinedValueTypeType const*>(expressionType))
```

Comment:
Do we need a special case here? Isn't this handled by `isExplicitlyConvertibelTo`?